src/modules/claim/services/claim-verification.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+import { Logger } from '../../logger/logger.service';
+import { IssuerVerificationService } from './issuer-verification.service';
+import { DIDService } from '../../did/did.service';
+import { RoleCredentialResolver } from '../resolvers/credential.resolver';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { RoleEIP191JWT, isEIP191Jwt } from '@energyweb/vc-verification';
+import { PreconditionType } from '@energyweb/credential-governance';
+
+@Injectable()
+export class ClaimVerificationService {
+  credentialResolver: RoleCredentialResolver;
+  constructor(
+    private readonly logger: Logger,
+    private readonly didService: DIDService,
+    private readonly issuerVerificationService: IssuerVerificationService
+  ) {
+    this.logger.setContext(ClaimVerificationService.name);
+    this.credentialResolver = new RoleCredentialResolver(didService);
+  }
+
+  /**
+   * Resolve a credential from storage and verify its proof/signature and its issuer's authority
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return void. Returns boolean indicating if credential is verified. Contains array of error messages if not verified.
+   */
+  public async resolveCredentialAndVerify(
+    subjectDID: string,
+    roleNamespace: string
+  ) {
+    const resolvedCredential = await this.credentialResolver.getCredential(
+      subjectDID,
+      roleNamespace
+    );
+    if (!resolvedCredential) {
+      return {
+        isVerified: false,
+        errors: [
+          `No credential found for role ${roleNamespace} for Did ${subjectDID}`,
+        ],
+      };
+    }
+    if (isEIP191Jwt(resolvedCredential)) {
+      return this.verifyRoleEIP191JWT(resolvedCredential as RoleEIP191JWT);
+    }
```

Comment:
I think we should have a more expected return here, rather than falling through to `undefined` 
```suggestion
    if (isEIP191Jwt(resolvedCredential)) {
      return this.verifyRoleEIP191JWT(resolvedCredential as RoleEIP191JWT);
    }
    return {
      isVerified: false,
      errors: [
        `credential found for role ${roleNamespace} but unable to verify for Did ${subjectDID}`,
      ],
    }
```

Reply:
@jrhender This will return the errors from verifyRoleEIP191JWT, which has the isVerifiedMessage and the errors, and the reason. return this.verifyRoleEIP191JWT(resolvedCredential as RoleEIP191JWT) returns
 return {
      errors: errors,
      isVerified: !!proofVerified && issuerVerified && !isExpired,
    };

Reply:
Sorry, I don't quite get your point. `this.verifyRoleEIP191EWT` is only called if `isEIP191Jwt` is true. If `isEIP191Jwt` is `false`, then this will just return `undefined`, I believe. It's this return of undefined that I think is best to avoid so that the return object is more consistent. Otherwise, I think we need to change the return type of the method to `Promise<{ isVerified: boolean; errors: string[] } | undefined>`.

Reply:
OKay, should the error then indicate that an unsupported credential type was resolved (i.e. not EIP191?) (indicating that a credential was found but we don't support verification methods for it?) @jrhender

Reply:
Just thinking that 'unable to verify for {did}' would make me think theres an issue with my DID rather than the credential itself, but maybe I'm overthinking it.

Reply:
I think something like `verification not supported for present credential type` or `No EIP191 credential found for the given namespace` should be more clear.

Reply:
@whitneypurdum yes, I think either of @nichonien's suggestions would be good

Reply:
@jrhender This has been udpated.

Reply:
Thanks, looks good